llvm/include/llvm/Support/AMDGPUAddrSpace.h

```diff
@@ -0,0 +1,31 @@
+//===---------------- AMDGPUAddrSpace.h -------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+/// \file
+/// AMDGPU address space definition
+///
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_AMDGPUADDRSPACE_H
+#define LLVM_SUPPORT_AMDGPUADDRSPACE_H
+
+namespace llvm {
+namespace AMDGPU {
+enum class AddrSpace {
```

Comment:
You can explicitly say `enum class AddrSpace : unsigned { ...` to avoid the need for the static_casts to unsigned.

Reply:
So this is still a second copy of the address space enum, another copy of which exists in AMDGPU.h (which does not use enum class, and uses different names). What's the plan to consolidate these?

Reply:
@arsenm I can consolidate these enums but I would like to be sure that I am allowed to do it. I was not sure why Clang enum has smaller range in comparison to LLVM enum.
My initial aim of this patch was to perform minimal code refactoring so that Clang and Flang can reuse the same enum. If you wish I can consolidate Flang, Clang and LLVM enums.

Reply:
The clang enum is very OpenCL centric and ignores AMDGPU and graphics specific address spaces. I don't see a point in keeping two overlapping enums

Reply:
@arsenm done. I unified the clang enum with LLVM enum. I haven't modified the names of LLVM enum and I haven't introduced `enum class` instead of `AMDGPUAS namespace`. Please let me know if is ok for you.